### PR TITLE
[FIX] Type Exception when writing res.partner.relation.all model

### PR DESCRIPTION
**Pull request by antespi**
Fix https://github.com/OCA/partner-contact/issues/129

### CODE REVIEW

File: partner_relations/model/res_partner_relation_all.py [hunk 1]
```diff
@@ -209,4 +210,6 @@ def create(self, vals):
     @api.one
     def unlink(self):
         """divert non-problematic creates to underlying table"""
-        return self.get_underlying_object().unlink()
+        underlying_objs = self.get_underlying_object()
+        underlying_objs = underlying_objs and underlying_objs[0]
+        return underlying_objs.unlink()
```
Comment (pedrobaeza) [inline partner_relations/model/res_partner_relation_all.py]: Why only remove the first one?

File: partner_relations/model/res_partner_relation_all.py [hunk 2]
```diff
@@ -177,6 +177,7 @@ def onchange_type_selection_id(self):
     def write(self, vals):
         """divert non-problematic writes to underlying table"""
         underlying_objs = self.get_underlying_object()
+        underlying_objs = underlying_objs and underlying_objs[0]
```
Comment (pedrobaeza) [inline partner_relations/model/res_partner_relation_all.py]: You can avoid this sentence putting at the end of the upper one `[:1]`

File: partner_relations/model/res_partner_relation_all.py (hunk 1 above)
Comment (antespi) [inline partner_relations/model/res_partner_relation_all.py]: This is an @api.one method

File: partner_relations/model/res_partner_relation_all.py (hunk 2 above)
Comment (antespi) [inline partner_relations/model/res_partner_relation_all.py]: But self.get_underlying_object() could return False, isn't it?

File: partner_relations/model/res_partner_relation_all.py (hunk 1 above)
Comment (antespi) [inline partner_relations/model/res_partner_relation_all.py]: After I've got a shower and re-thought what you wrote, you,re rigth. Here we need a loop or make self.get_underlying_object() returns a recordset not a list.

File: partner_relations/model/res_partner_relation_all.py (hunk 2 above)
Comment (antespi) [inline partner_relations/model/res_partner_relation_all.py]: Same that other coment: make self.get_underlying_object() returns a recordset not a list.

File: partner_relations/model/res_partner_relation_all.py (hunk 1 above)
Comment (adrienpeiffer) [inline partner_relations/model/res_partner_relation_all.py]: Don't hesitate to correct me if i'm wrong but `self.get_underlying_object()` is call in an @api.one method.
Then if `get_underlying_object` use `self.id`. This method should return a list with only 1 object.

**Comment by adrienpeiffer** [comment]
Why not change the decorator of `get_underlying_object()` to avoid to have a list of recordset. We can change this to `@api.model` and add id in the method declaration. What do you think ?

**Comment by antespi** [comment]
@adrienpeiffer you're right. Remove api.one is better solution. I've marked `get_underlying_object()` method as private, it's used only in this model as internal method.

**Comment by antespi** [comment]
Please @bwrsandman, could you check this bugfix?

**Comment by bwrsandman** [comment]
Looks fine, but I am no longer doing any Odoo.

**Comment by rafaelbn** [comment]
Looks fine :+1:

**Comment by pedrobaeza** [comment]
The method is still unprotected from being called from a multi-recordset, raising the ensure_one in `self.id`, but I see that this method is only called from `@api.one` methods, so :+1:

**Comment by adrienpeiffer** [comment]
:+1:

**Comment by antespi** [comment]
@bwrsandman We'll miss you so much. Good luck wherever you go.

**Pull request merged**